Add more factory methods

julienrf opened the pull request:
Fixes #169.

Ichoran commented on src/main/scala/strawman/collection/Factories.scala:
```diff
@@ -36,14 +38,146 @@ trait CanBuild[-A, +C] extends Any with BuildFrom[Any, A, C] {
+    *  @param   elem the element computation
+    *  @return  A $coll that contains the results of `n1 x n2` evaluations of `elem`.
+    */
+  def fill[A](n1: Int, n2: Int)(elem: => A): CC[CC[A] @uncheckedVariance] = tabulate(n1)(_ => fill(n2)(elem))
```
Why does this defer to `tabulate` instead of `fill`?

Ichoran commented on src/main/scala/strawman/collection/View.scala:
```diff
@@ -69,18 +69,22 @@ object View extends IterableFactory[View] {
-        }
-      }
+    def iterator() = Iterator.fill(n)(elem)
     override def knownSize: Int = n
```
This should be fixed to be `0 max n` like the others.

Ichoran commented:
What's there looks good (with a couple tweaks I suggested).  But tests would help us be more confident!

julienrf commented:
@Ichoran I applied your suggestions and added some tests.

I also moved the `tabulate` and `fill` factory methods to a new `SeqFactory` trait because these methods are only relevant for `Seq` collections (they make no sense on `Set`, for instance).

Ichoran commented:
LGTM
